### ISSUE TITLE
confusion in how convert finetuned ckpts to HF format

### ISSUE THREAD

**Issue by optimass** [pytorch/torchtune#958]
Hello.
Sorry, I know there are a couple of open issues about this, but I'm still quite confused about how to convert finetuned ckpts to Huggingface format, and what would be the best practice.

Here's what I understand.
- if you use the `FullModelMetaCheckpointer` --> `meta_model_{i}.pt`
- if you used the `FullModelHFCheckpointer` --> `hf_model_{idx}_{i}.pt` 

I haven't found a simple way to convert these to HF format.
For context, I have followed https://github.com/pytorch/torchtune/issues/931 which led me nowhere.

Thanks!

**Comment by kartikayk** [comment]
Thanks for opening this issue!

I think there are two different conversations here as I understand it:
1) Convert into HF format - this is already done by the FullModelHFCheckpointer (this assumes the input checkpoint is also in HF format)
2) Use with ```from_pretrained``` API - this actually needs us to have the configs/json files pointing to the right checkpoint names.

I think we need to figure out 2) a bit. 1) should work, but let me know if you observe something different.

I

**Comment by optimass** [comment]
1. yes you are right!
2. I've manage use `from_pretrained`. I needed to cp some config files + rename the `hf_model_0001_0.pt` --> `pytorch_model-00001-of-00004.bin`. (Maybe there could be some utils to automate this for the supported models?)

Now, somehow, when try to instantiate a TGI server using the converted ckpts, I get an error that TGI is looking for an `adapter.json` (seems like it's assuming the finetuned model is a PEFT one...)
Anyhow, this is out of scope.
Closing the issue!
